catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImplTest.java

```diff
@@ -13,6 +13,1240 @@
+
+    @Mock
+    private ExpressionBuilder expressionBuilder;
+
+    @Mock
+    private Subject subject;
+
+    @Mock
+    private BundleContext bundleContext;
+
+    @Mock
+    private ServiceReference serviceReference;
+
+    @Mock
+    private RegistryStore registryStore;
+
+    @Mock
+    private DeleteResponse deleteResponse;
+
+    @Mock
+    private UpdateResponse updateResponse;
+
+    @Mock
+    private CreateResponse createResponse;
+
+    @Before
+    public void setUp() throws Exception {
+        when(parser.configureParser(anyList(), any(ClassLoader.class))).thenReturn(configurator);
+        federationAdminServiceImpl = spy(new FederationAdminServiceImpl(security));
+        federationAdminServiceImpl.setRegistryTransformer(registryTransformer);
+        federationAdminServiceImpl.setCatalogFramework(catalogFramework);
+        federationAdminServiceImpl.setParser(parser);
+        federationAdminServiceImpl.setFilterBuilder(filterBuilder);
+        System.setProperty(SystemInfo.SITE_NAME, TEST_SITE_NAME);
+        System.setProperty(SystemInfo.VERSION, TEST_VERSION);
+        testMetacard = getPopulatedTestRegistryMetacard();
+        when(deleteResponse.getProcessingErrors()).thenReturn(new HashSet<ProcessingDetails>());
+        when(createResponse.getProcessingErrors()).thenReturn(new HashSet<ProcessingDetails>());
+        when(updateResponse.getProcessingErrors()).thenReturn(new HashSet<ProcessingDetails>());
+        when(catalogFramework.update(any(UpdateRequest.class))).thenReturn(updateResponse);
+        when(catalogFramework.delete(any(DeleteRequest.class))).thenReturn(deleteResponse);
+        when(catalogFramework.create(any(CreateRequest.class))).thenReturn(createResponse);
+        when(security.runWithSubjectOrElevate(any(Callable.class))).thenAnswer(invocation -> {
+            try {
+                return ((Callable) invocation.getArguments()[0]).call();
+            } catch (Exception e) {
+                throw new InvocationTargetException(e.getCause());
+            }
+        });
+
+    }
+
+    @Test
+    public void testGetRegistryObjectByMetacardId() throws Exception {
+        String metacardId = TEST_METACARD_ID;
+        Metacard metacard = testMetacard;
+        QueryRequest request = getTestQueryRequest();
+        QueryResponse response = getPopulatedTestQueryResponse(request, metacard);
+        JAXBElement<RegistryPackageType> jaxbRegistryPackage =
+                EbrimConstants.RIM_FACTORY.createRegistryPackage(getTestRegistryPackage());
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response);
+        when(parser.unmarshal(any(ParserConfigurator.class),
+                eq(JAXBElement.class),
+                any(InputStream.class))).thenReturn(jaxbRegistryPackage);
+        RegistryPackageType registryPackage =
+                federationAdminServiceImpl.getRegistryObjectByRegistryId(metacardId, null);
+        assertThat(registryPackage, notNullValue());
```

Comment:
We could be more specific here and assign the return value of `getTestRegistryPackage()` to a local variable, say `expectedRegistryPackage`, and do `assertThat(registryPackage, is(sameInstance(expectedRegistryPackage))`.